parts/k8s/kubernetesmastercustomdata.yml

```diff
@@ -182,23 +182,13 @@ MASTER_ARTIFACTS_CONFIG_PLACEHOLDER
 {{end}}
 
 {{ if .HasAadProfile }}
-    OIDC_CLIENT_ID=spn:{{WrapAsVariable "aadServerAppId"}}
     VAR_AAD_TENANT_ID={{WrapAsVariable "aadTenantId"}}
     VAR_TENANT_ID={{WrapAsVariable "tenantId"}}
-    VAR_TARGET_ENV={{WrapAsVariable "targetEnvironment"}}
     AAD_TENANT_ID=${VAR_AAD_TENANT_ID:-$VAR_TENANT_ID}
-    AAD_ISSUER_HOST="sts.windows.net"
-    if [ "$VAR_TARGET_ENV" = "AzureChinaCloud" ]; then
-        AAD_ISSUER_HOST="sts.chinacloudapi.cn"
-    fi
-
-    OIDC_ISSUER_URL="https://$AAD_ISSUER_HOST/$AAD_TENANT_ID/"
-    perl -pi -e "s|--oidc-client-id=\K(?=\")|$OIDC_CLIENT_ID| || s|--oidc-issuer-url=\K(?=\")|$OIDC_ISSUER_URL|" "/etc/kubernetes/manifests/kube-apiserver.yaml"
-{{else}}
-    sed -i "/--oidc-client-id\|--oidc-issuer-url\|--oidc-username-claim/d" "/etc/kubernetes/manifests/kube-apiserver.yaml"
+    sed -i "/--oidc-issuer-url/s/$/$AAD_TENANT_ID/" "/etc/kubernetes/manifests/kube-apiserver.yaml"
```

Comment:
We still need to do this post-deployment transformation because I'm not aware of any other way to implement the fallback tenantID assignment. The one here uses this `kubernetesmastervars.t` statement:

`"tenantId": "[subscription().tenantId]",`

Which gives us our fallback `tenantId` value (see `AAD_TENANT_ID=${VAR_AAD_TENANT_ID:-$VAR_TENANT_ID}` above). My understanding is that `subscription()` is an ARM function that we won't have access to anywhere else.